src/platform/plugins/shared/discover/public/application/main/state_management/discover_saved_search_container.ts

```diff
@@ -176,18 +181,38 @@ export function getSavedSearchContainer({
 
   const persist = async (nextSavedSearch: SavedSearch, saveOptions?: SavedObjectSaveOpts) => {
     addLog('[savedSearch] persist', { nextSavedSearch, saveOptions });
+
+    const dataView = nextSavedSearch.searchSource.getField('index');
+    const profileDataViewIds = internalStateContainer.getState().defaultProfileAdHocDataViewIds;
+    let replacementDataView: DataView | undefined;
+
+    // If the Discover session is using a default profile ad hoc data view,
+    // we copy it with a new ID to avoid conflicts with the profile defaults
+    if (dataView?.id && profileDataViewIds.includes(dataView.id)) {
+      replacementDataView = await services.dataViews.create({
+        ...dataView.toSpec(),
+        id: uuidv4(),
+        name: i18n.translate('discover.savedSearch.defaultProfileDataViewCopyName', {
+          defaultMessage: '{name} (copy)',
```

Comment:
Let's use the saved search name instead of `copy`. Otherwise it's getting out of control why do I have an increasing number of temporary data views after saving multiple times.

<img width="360" alt="Screenshot 2025-01-23 at 13 23 47" src="https://github.com/user-attachments/assets/146a325d-545b-4ee9-b2b5-d04abbafa0ff" />